src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -247,6 +250,119 @@ public static RecordValue Execute()
             }
         }
 
+        // Must have "Function" suffix. 
+        private class TestInvalidRecordCustomFunction : ReflectionFunction
+        {
+            public TestInvalidRecordCustomFunction()
+                : base(
+                      "InvalidRecordTest",
+                      RecordType.Empty().Add(new NamedFormulaType("num", FormulaType.Number)))
+            {
+            }
+
+            // Must have "Execute" method. 
+            public static RecordValue Execute()
+            {
+                var recordType = RecordType.Empty()
+                    .Add(new NamedFormulaType("num1", FormulaType.Number));
+                var val = FormulaValue.NewRecordFromFields(recordType, new NamedValue("num1", FormulaValue.New(1)));
+                return val;
+            }
+        }
+
+        [Theory]
+        [InlineData("({x: 2})", true)]
+        [InlineData("({x: 2, y: 5})", true)]
+        [InlineData("({x: \"2\", y: \"5\"})", true)]
+        [InlineData("({x: 2, y: 5, a: 8})", false)]
+        [InlineData("({a: 8})", false)]
+        [InlineData("({x: {x : 2}})", false)]
+        public void CustomFunctionRecordTest(string inputRecord, bool isSuccess)
+        {
+            var record = RecordType.Empty().Add(new NamedFormulaType("x", FormulaType.Number)).Add(new NamedFormulaType("y", FormulaType.Number));
+            CustomFunctionRecord(new TestRecordsCustomFunction(record), inputRecord, isSuccess);
+        }
+
+        [Theory]
+        [InlineData("({x: 2})", true)]
+        [InlineData("({x: 2, y: 5})", false)]
+        public void CustomFunctionNonRecordTest(string inputRecord, bool isSuccess)
```

Comment:
>CustomFunctionNonRecordTest [](http://example.com/codeflow?start=20&length=27)

How is this test different than the previous one? The first case seems identical to the second case in the previous test (same schema expected / actual), and the second one is the same as the fourth case in the previous test (one extra property in the actual)

Reply:
This test different custom function where resulttype is not record. same input record. Just want to test that this work for both result type is record or not.